### ISSUE TITLE
[Favourite Media] Ability to create category from with Context Menu

### ISSUE THREAD

**Issue by Aezaeo** [Dastan21/BDAddons#71]
When right-clicking a media file we can send that file to a specific category from within the FavoriteMedia context menu, but, in order to create a category we have to open FavoriteMedia and right-click in the open space. It would be convenient and more efficient to have the ability to create a category for a media file through the context menu as well.

For example, you find an image that doesn't belong to any of the previously created categories, so instead of having to open FavoriteMedia via the buttons within the text area to create the category, a user would be able to do it just by using the context menu.

Adding the ability to add media to FavoriteMedia via context menu, imo, is one of it's better enhancements. This suggestion is just to add to that. Also, this idea fixes having to load up all your favorites, which can have some load time if you have a lot of media favorited, just to create a new category.

Additional Enhancement request(s):

I recently created my own server to copy media links that I like/favorite from other servers to my server for personal collection and to share with my friends in my server. Much like what FavoriteMedia is already capable of doing. I understand FavoriteMedia has the option to multi-post via holding shift and clicking multiple media files within the FavoriteMedia list to upload to my server. 

The issue comes when I get asked for the source to an image and I have to go searching for the (twitter,instgram,i.e.) link from the server I found the image in to begin with. This requires a lot of server switching (copy/pasting links) just to alleviate this issue. It would be extremely convenient if I could use FavoriteMedia to send a media file/link/post directly to a specified channel of my choosing within my server located from within a different server. 

Example:
I find a cool art post on a different server that is a twitter link, directly from the twitter user responsible for the image. I favorite the image via FavoriteMedia to upload to my server for collection purposes and to share with my friends. We then want to see more of this twitter persons art so I have to additionally add the twitter link. If FavoriteMedia allowed me to just the send the link/post to my server/channel directly via context menu I could avoid having to copy/paste a link/post to my server. 

I also noticed that one can only download all media if the media is already within a category. There is no option to download all for unsorted media. It would be nice if one could download all unsorted media  as well. For some us lazy users who haven't taken the time to categorize their media but wish to download all their favortied media anyway.

I've said this before but FavoriteMedia is my most used plugin, basically my only reason for sticking with BetterDiscord. So I constantly experience things that I wish were added to the plugin. I appreciate the updates and your time, thanks.

**Comment by Dastan21** [comment]
Thanks for your detailed suggestions :D
I might add the first and last ones.

About the "source" suggestion, I don't really understand what you suggest, is it to save links which are not media ? But where would they be saved ? With the media link ?

**Comment by Aezaeo** [comment]
> Thanks for your detailed suggestions :D I might add the first and last ones.
> 
> About the "source" suggestion, I don't really understand what you suggest, is it to save links which are not media ? 
Basically, yes.

For instance, someone posts a twitter post which has an image/video but the twitter link itself has the twitter user's name whereas none of the image/videos do. Now, with FavoriteMedia one can favorite the media itself and change the filename to match the twitter user's name, which in and of itself is a very good feature. But having to go through multiple images/videos to change the filename can become tedious for a user with a lot of images/videos. 

So, instead of having to use that process. One could just "send" the link to a specific server/channel to be posted. All this bein done within the context menu. 

But where would they be saved ? 
They'd technically just be copied over to a specified Server(channel).

With the media link ?
Yes.

Currently, FavoriteMedia only allows you to save(favorite) a media file then use the upload function to add the media files to a server. This current method eliminates the "Source" link which could have information such as the twitter handle for a specified image, which provides more content related images from  that "source" link. 

Being able to send links/media from one server to another directly without having to save(favorite) the media file then upload to a different server just removes the extra steps. Instead of using the FavoriteMedia upload function, one could just right click a "link/media" choose send to Server X-Channel X and that post would basically be copied and posted into the server/channel that was chosen. Unlike copy message link which only provides a link to the post itself.

I got this idea from a program I use to scrape discord servers for images. The dev recently added a feature to send files from the server being scraped to your own server. This feature was added for people who want to save on hard drive space. So instead of scraping and downloading 400k images/videos from a server to your hard drive, the user now has the option to use a discord server as their storage space.

Here's the link to the devs github commit. 
[https://github.com/get-got/discord-downloader-go/commit/7c0afe0df82e490f7a9b08f905ad834171ff6986](url)

Maybe, this will help you better understand what I'm suggesting. Anything you decide to I will respect. I just thought I'd share some ideas for enhancing FavoriteMedia. Thanks again, for your time.
